docs/dataprovider.rst

```diff
@@ -0,0 +1,35 @@
+DataProvider
+----------------
+
+An important part of any algorithm is the data its working over and the data that it produces.
```

Comment:
```suggestion
An important part of any algorithm is the data it's working over and the data that it produces.
```

Reply:
Note on suggestions: Github lets these be separate commits on the branch. I would personally prefer that these get fixup/squashed into the relevancy commit that made the original change here. If that provides to be an issue, I won't block this PR however.

Reply:
Yes I also realize that I am the original offender that wrote the wrong things here. :) But we might as well fix them while we're here.